src/mediaLibrary/MediaLibraryVideoClient.ts

```diff
@@ -0,0 +1,217 @@
+   */
+  getPlaybackInfo(
+    assetIdentifier: MediaLibraryAssetInstanceIdentifier,
+    options: MediaLibraryPlaybackInfoOptions = {},
+  ): Observable<VideoPlaybackInfo> {
+    const {instanceId, libraryId} = parseAssetInstanceId(assetIdentifier)
+    const uri = buildVideoPlaybackInfoUrl(instanceId, libraryId)
+    const queryParams = buildQueryParams(options)
+
+    return _request<VideoPlaybackInfo>(this.#client, this.#httpRequest, {
+      method: 'GET',
+      uri,
+      query: queryParams,
+    })
+  }
+
+  /**
+   * Extract playback tokens from signed video playback info
+   *
+   * @param playbackInfo - The playback info response
+   * @returns Object containing playback, thumbnail, and storyboard tokens if signed, undefined otherwise
+   */
+  getPlaybackTokens(playbackInfo: VideoPlaybackInfo): VideoPlaybackTokens | undefined {
+    if (isSignedPlaybackInfo(playbackInfo)) {
+      return {
+        playback: playbackInfo.stream.token,
+        thumbnail: playbackInfo.thumbnail.token,
+        storyboard: playbackInfo.storyboard.token,
+        animated: playbackInfo.animated.token,
+      }
+    }
+
+    return undefined
+  }
+}
+
+/** @internal */
+export class MediaLibraryVideoClient {
+  #client: SanityClient
+  #httpRequest: HttpRequest
+  constructor(client: SanityClient, httpRequest: HttpRequest) {
+    this.#client = client
+    this.#httpRequest = httpRequest
+  }
+
+  /**
+   * Get video playback information for a media library asset
+   *
+   * @param assetIdentifier - Asset instance identifier (GDR, video-prefixed ID, or container ID)
+   * @param options - Options for transformations and expiration
+   */
+  getPlaybackInfo(
+    assetIdentifier: MediaLibraryAssetInstanceIdentifier,
+    options: MediaLibraryPlaybackInfoOptions = {},
+  ): Promise<VideoPlaybackInfo> {
+    return lastValueFrom(
+      new ObservableMediaLibraryVideoClient(
+        this.#client.observable,
+        this.#httpRequest,
+      ).getPlaybackInfo(assetIdentifier, options),
+    )
+  }
+
+  /**
+   * Extract playback tokens from signed video playback info
+   *
+   * @param playbackInfo - The playback info response
+   * @returns Object containing playback, thumbnail, and storyboard tokens if signed, undefined otherwise
+   */
+  getPlaybackTokens(playbackInfo: VideoPlaybackInfo): VideoPlaybackTokens | undefined {
+    if (isSignedPlaybackInfo(playbackInfo)) {
+      return {
+        playback: playbackInfo.stream.token,
+        thumbnail: playbackInfo.thumbnail.token,
+        storyboard: playbackInfo.storyboard.token,
+        animated: playbackInfo.animated.token,
+      }
+    }
+
+    return undefined
+  }
+}
+
+function parseAssetInstanceId(assetIdentifier: MediaLibraryAssetInstanceIdentifier): {
+  instanceId: string
+  libraryId?: string
+} {
+  if (typeof assetIdentifier === 'string') {
+    // Handle video-prefixed asset instance ID
+    if (assetIdentifier.startsWith('video-')) {
+      return {instanceId: assetIdentifier}
+    }
```

Comment:
Is this check redundant if the returned value is the same as the fallback?

Reply:
Good point. I think I want to reorder this logic a bit, so that it either checks for a `video-` prefixed instance id string or continues with the other logic.